### ISSUE TITLE
filler

### ISSUE THREAD

**Issue by pr11t** [01-edu/public#1481]
The game_engine sometimes provides an invalid piece to a player.

For example with the following seed player 1 will always lose because the piece cannot be placed.
`./game_engine -f maps/map01 -p1 robots/terminator -p2 robots/wall_e -seed 1664821796665768000`

**Comment by mikysett** [comment]
Hello pr11t, thanks a lot for your feedback and for the example with the seed.

This is a scenario we are aware of and this is why during audits we never ask a 100% success rate.
Those edge cases are a normal consequence of implementing a randomly generated piece engine, sometimes it will give players pieces hard to place at the very start of the game.

Also to notice a little Easter egg we leaved for students to discover... Actually **it is** possible for the robots to place even that first piece!

Nothing prevents you from giving to the game engine negative positions as an answer... For example in this case a correct answer could be `4 -1`, the game engine is perfectly capable to handle this.

The reason why none of our robots implements it is to prevent them from having this extra advantage against students that do not find this little trick.

So to recap:
- No need to have 100% success rate to pass the audits
- If you want to go the extra mile there is a way to program a robot able to take advantage of those edge cases

The implementation is not so trivial due to Rust type safety checks, I hope you will go the extra mile and having fun implementing that feature.

**Comment by pr11t** [comment]
Thank you for the thorough explanation :)
